aviary/subsystems/propulsion/propeller_map.py

```diff
@@ -0,0 +1,143 @@
+        self.data = {key: np.array([]) for key in PropModelVariables}
+
+        # Create dict for variables present in propeller data with associated units
+        self.propeller_variables = {}
+
+        data_file = options.get_val(Aircraft.Engine.PROPELLER_DATA_FILE)
+        self._read_data(data_file)
+
+    def _read_data(self, data_file):
+        # read csv file
+        raw_data = read_data_file(data_file, aliases=aliases)
+
+        message = f'<{data_file}>'
+        # Loop through all variables in provided data. Track which valid variables are
+        #    included with the data and save raw data for reference
+        for key in get_keys(raw_data):
+            val, units = raw_data.get_item(key)
+            if key in aliases:
+                # Convert data to expected units. Required so settings like tolerances
+                # that assume units work as expected
+                try:
+                    val = np.array([convert_units(i, units, default_prop_units[key])
+                                   for i in val])
+                except TypeError:
+                    raise TypeError(f"{message}: units of '{units}' provided for "
+                                    f'<{key.name}> are not compatible with expected units '
+                                    f'of {default_prop_units[key]}')
+
+                # propeller_variables currently only used to store "valid" engine variables
+                # as defined in EngineModelVariables Enum
+                self.propeller_variables[key] = default_prop_units[key]
+
+            else:
+                if self.get_val(Settings.VERBOSITY).value >= 1:
+                    warnings.warn(
+                        f'{message}: header <{key}> was not recognized, and will be skipped')
+
+            # save all data in self._original_data, including skipped variables
+            self._original_data[key] = val
+
+        if not self.propeller_variables:
+            raise UserWarning(
+                f'No valid propeller variables found in data for {message}')
+
+        # Copy data from original data (never modified) to working data (changed through
+        #    sorting, generating missing data, etc.)
+        # self.data contains all keys in PropellerModelVariables
+        for key in self.data:
+            self.data[key] = self._original_data[key]
+
+    def read_and_set_mach_type(self, data_file):
+        # sample header:
+        # created 06/24/24 at 17:13
+        # GASP propeller map converted from PropFan.map
+        # Propfan format - CT = f(Mach, Adv Ratio & CP)
+        # mach_type = mach
+        # Hamilton Standard 10 Bladed Propfan Performance Deck:  Ct Tables
+
+        fp = get_path(data_file)
+        with open(fp, "r") as f:
+            line = f.readline()
+            line = f.readline()
+            line = f.readline()
+            line = f.readline()  # read 4th line
+            mach_type = line.split()[3]  # get token 3
+        return mach_type
```

Comment:
@crecine , Can you be more specific? I don't quite understand what I need to do.

Reply:
Currently, you are reading the comments (lines that start with `#`) and using data from that.
Typically, comments are not required, but are just extra information for clarification.
As is, the file must have the value for mach_type in the fourth row and 3rd column.

I think it would be better to use a while loop here to start reading lines, removing comments, and discarding empty lines. Then capture the first non comment line and read data from that.
Something like this, but you might need to fix it up a bit:
```
        with open(fp, "r") as f:
            line = f.readline()
            while len(line.split('#')[0].strip())==0:
                line = f.readline()
            mach_type = line.split('#')[0].split('=')[-1].strip()
```
this will:
open the file
read the first line
get rid of anything after a `#`, remove leading and trailing whitespace, and then check the length
if the length is 0, read another line (we are still in the header and have no useful data
if the length is non-zero, we have data we intended to read
discard any comments, split the line at the equal sign and get the value

we can then have a sample header like this:
```
# created 06/24/24 at 17:13
# GASP propeller map converted from PropFan.map
# Propfan format - CT = f(Mach, Adv Ratio & CP)
# Hamilton Standard 10 Bladed Propfan Performance Deck:  Ct Tables
mach_type = mach
```

 but the following would also produce a valid deck:
```
# We got this data from Xfoil
mach_type=mach # this means we want to use freestream Mach, not helical mach
```

Reply:
If I move `mach_type` out from comment block, csv reader will fail.